test/route.4.test.js

```diff
@@ -34,18 +33,18 @@ test('route error handler overrides global custom error handler', t => {
     errorHandler: customRouteErrorHandler
   })
 
-  fastify.inject({
-    method: 'GET',
-    url: '/more-coffee'
-  }, (error, res) => {
-    t.error(error)
-    t.equal(res.statusCode, 418)
-    t.same(JSON.parse(res.payload), {
+  {
```

Comment:
Why are we wrapping things in blocks?

Reply:
I deleted this block that was useless, the other ones have the conflicting variables name. 

As for the othe PR if you have any suggestion I will update it!